### ISSUE TITLE
[Question] Design considerations for SCER spec/label

### ISSUE THREAD

**Issue by chrisxie-fw** [Green-Software-Foundation/scer#90]
### Question

Here, I'm taking the liberty to share some of the key design principles of the SCER spec that I 've personally been trying to follow:
- **Business value**: how would companies see business value in adopting SCER spec/label? There are several ways where SCER can present business value:
    - **Dollar amount, in terms of Social Cost of Carbon (SCC)**. This is listed as one of the options of carbon accounting measurements in [the benchmark section of the original SCER spec](https://github.com/Green-Software-Foundation/scer/blob/Dev/Software_Carbon_Efficiency_Rating(SCER)/Software_Carbon_Efficiency_Rating_Specification.md#22-benchmarking-specification). Some analogy can be found in something like the energy guide that spells out the dollar amount of an appliance. Most people understand dollars more than CO2g per software function. 

![EnergyStar](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTdSPRhyBWxJDc5KBj5wLKKSL3gyoCtgMNBmyc_M4ErKJ5xF10SDOJxr5VHRfS4NdGdbhc&usqp=CAU
)

This is where the original SCER spec leaves room for flexibility to cater for different needs, because some may prefer SCI, CO2ge,SCC, or other measurement units. For example, huggingface uses [Energy (tokens/kWh)](https://huggingface.co/spaces/optimum/llm-perf-leaderboard), which could correlate to carbon efficiency because they use their own internal uniform testing/benchmarking environment.  

   - **Compliance to existing policy mandate**: companies that comply with existing climate mandate presents a competitive advantage in the marketplace than those who don't. Linking SCER to policy is a policy incentive for companies to adopt SCER. 

- **Ease of use**: the spec is extremely easy for people to follow, how to keep a balance between spec's rigidness/ completeness/comprehensiveness v.s. simplicity and ease of adoption (easy to understand and follow and implement the spec, etc). Personally in my mind, the food label NutriScore is a great example of striking the right balance between information completeness and intuitive simplicity. As a consumer, I can very easily trust and follow the NutriScore label in making food purchasing decisions. 
![Nutri-Score](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTaSK-xId-y3XF350rY_AtSc0BltkUEcAQrv7AOEiBnQ1i2w97nXNP5PcYidfHqlARDwTo&usqp=CAU)

- **With the END in mind, working backwards**:
    - Who are the end users of SCER spec? Maybe software developers, product managers, business executives (VPs), company CEOs, and general consumers of software products and services etc. How would SCER spec appeal to these different personas? 
        - How could business executives and CEOs, decide to support and adopt SCER? Would displaying SCC dollar amounts help them in making those decisions?
        - How could developers and product managers easily obtain/generate a SCER label?
        - How would SCER label enable general consumers to make intuitive choices of software products and services?
    - How do we envision people to use the SCER spec/label?
        - How will users generate the SCER label?
        - Are there open source tools available to compute SCI scores or SCC dollars? 
- **Regarding conformance, verification**:
    - What would be an appropriate mechanism for SCER label conformance and verification? 
    - Do we assume people will intentionally abuse the label? 
    - Vienna's Underground/Metro system adopts more of an honoring system - there is nobody to check at the entrance whether a passenger buys the ticket or not. But if s/he gets caught without a ticket, it could be a costly affair. On the contrary, many large metropolitans like Paris will have ticket offices and gates to enter the Metro. I'm guessing that Vienna’s honor system is more of a reflection of a community-oriented, high-trust approach to public transportation, supported by a cultural context that values civic responsibility. Larger cities like Paris, with different socio-economic conditions, transit scale, and security considerations, need more rigid systems to maintain efficiency and revenue integrity. In the context of SCER label conformance and verification, are we a Vienna or a Paris? 
    - As an open source project or an open standard project, there is limited resources to enforce adherance to the spec/label requirements. The current design of the [SCER Certification Program](https://github.com/Green-Software-Foundation/scer/tree/Dev/SCER_Certification_Program) is adopting a mechanism that's more like an honoring system. People who use SCER label are encouraged to report back their usage to a github page. As a reward, they get a SCER Conformance Label that they can put in their website to demonstrate their compliance which in turn is of business value. 
    - Are there open source and easy mechanisms to address the unfortunate case of intentional abuse of the SCER Label? For example, if someone intentionally misrepresent the Label and its data not matching up what's computed, would there be a public trusting score of that label, or if someone reported a label that is being misrepresented, would there be a status column in the [Report.md](https://github.com/Green-Software-Foundation/scer/blob/Dev/SCER_Certification_Program/Reporting.md) page that says this organization's use of the label is under investigation?

**Comment by jawache** [comment]
@chrisxie-fw 

Regarding verification, this is being explored heavily with impact framework. The most obvious way to verify is to just be fully transparent with all your data, that way it's trivial for the end user to be able to check. But we're also exploring ways to cryptography sign a manifest file so trusted 3rd parties can verify deeper. We should leverage the work being done there, as long as the evidence is a manifest file - the tools, technologies and agreements are already being explored and researched by other people.

I think we should make a business case white paper, we can even launch a workshop and invite members to participate.

However, In addition to that, I firmly believe for this label to be a success there needs to be a business around the label. I.e. lots of people need to be making a living **creating** a SCER label.
- Orgs who we give permission to certify manifest files and publish labels. These orgs will be incentivised to convince other orgs to get a SCER label for their software product.
- People who we have trained up on how to compute a SCI score and generate a manifest file. These people should be earning a living computing and publishing scer labels for the companies that hire them.

**Comment by chrisxie-fw** [comment]
I agree there needs to be a strong business case for companies to adopt the SCER label. At the minimum, when companies adopting the SCER label, the Label adds more trust and credibility of environmental consciousness to the company's products and services.

**Comment by adityamanglik** [comment]
I agree with Chris's design principles for the SCER spec. Here's my perspective on some of his questions:

1. Business Value

* Measurement Flexibility: Supporting multiple units (SCI, CO2ge, SCC) ensures adaptability across various industry standards and internal benchmarks, similar to EnergyStar. This broadens the label's applicability. However, the ease and accuracy of converting between these metrics are unclear. A reliable conversion process or simultaneous measurement of all metrics should be considered.

2. Ease of Use

* NutriScore Analogy: SCER maintains comprehensive metrics while ensuring labels are intuitive for quick decision-making, similar to NutriScore for food. Implementing QR codes for verification allows users to access detailed metrics when needed.

3. End-User Segmentation

* Executives and CEOs: Displaying SCC in dollars aids strategic decision-making and highlights commitment to sustainability.

* Developers and Product Managers: Providing open-source tools and APIs for automatic SCER label generation facilitates integration into existing workflows. This aspect will be a primary focus.

* Consumers: The label and QR codes should prioritize end-consumers, as they represent the largest user group for the labels.

4. Label Generation and Tools

* Open-Source Libraries: Developing open-source libraries for computing SCI scores and SCC dollars is underway, ensuring accessibility and standardization across implementations. Emphasizing this area is crucial.

* Transparent Reporting: Developing a new service to display data is necessary since reliance on other companies to adopt the label and metrics is not feasible. The ideal goal is a uniform metric and label style across different companies and industries over time.
